src/FsAutoComplete.Core/TipFormatter.fs

```diff
@@ -1042,82 +1091,199 @@ let formatPlainTip (ToolTipText tips) : (string * string) =
+            match computeGenericParametersText tooltipData with
+            | Some genericParametersText -> docCommentText + nl + nl + genericParametersText
+            | None -> docCommentText
+
+          consolidatedDocCommentText, xmlDoc.HasTruncatedExamples
+
+        | TryGetXmlDocMemberResult.None ->
+          // Even if a symbol doesn't have a doc comment, it can still have generic parameters
+          let docComment =
+            match computeGenericParametersText tooltipData with
+            | Some genericParametersText -> genericParametersText
+            | None -> ""
+
+          docComment, false
+        | TryGetXmlDocMemberResult.Error -> ERROR_WHILE_PARSING_DOC_COMMENT, false
+
+      {| DocComment = docComment
+         HasTruncatedExamples = hasTruncatedExamples |}
+      |> Ok
+      |> Some
+
+    | ToolTipElement.CompositionError error -> error |> Error |> Some
+
+    | ToolTipElement.Group []
+    | ToolTipElement.None -> None)
+
+/// <summary>
+/// Try format the given tooltip with the requested style.
+/// </summary>
+/// <param name="toolTipText">Tooltip documentation to render in the middle</param>
+/// <param name="formatCommentStyle">Style of tooltip</param>
+/// <returns>
+/// - <c>TipFormatterResult.Success {| DocComment; HasTruncatedExamples |}</c> if the doc comment has been formatted
+///
+///   Where DocComment is the format tooltip and HasTruncatedExamples is true if examples have been truncated
+///
+/// - <c>TipFormatterResult.None</c> if the doc comment has not been found
+/// - <c>TipFormatterResult.Error string</c> if an error occurred while parsing the doc comment
+/// </returns>
+let tryFormatTipEnhanced toolTipText (formatCommentStyle: FormatCommentStyle) =
+
+  match tryComputeTooltipInfo toolTipText formatCommentStyle with
+  | Some(Ok tooltipResult) -> TipFormatterResult.Success tooltipResult
+
+  | Some(Error error) -> TipFormatterResult.Error error
+
+  | None -> TipFormatterResult.None
+
+/// <summary>
+/// Generate the 'Show documentation' link for the tooltip.
+///
+/// The link is rendered differently depending on if examples
+/// have been truncated or not.
+/// </summary>
+/// <param name="hasTruncatedExamples"><c>true</c> if the examples have been truncated</param>
+/// <param name="xmlDocSig">XmlDocSignature in the format of <c>T:System.String.concat</c></param>
+/// <param name="assemblyName">Assembly name, example <c>FSharp.Core</c></param>
+/// <returns>Returns a string which represent the show documentation link</returns>
+let renderShowDocumentationLink (hasTruncatedExamples: bool) (xmlDocSig: string) (assemblyName: string) =
+
+  // TODO: Refactor this code, to avoid duplicate with DocumentationFormatter.fs
+  let content =
+    Uri.EscapeDataString(sprintf """[{ "XmlDocSig": "%s", "AssemblyName": "%s" }]""" xmlDocSig assemblyName)
+
+  let text =
+    if hasTruncatedExamples then
+      "Open the documentation to see the truncated examples"
```

Comment:
Does 'truncated' here mean that there were so many examples that we didn't show them all? If so, what about the following message:

```suggestion
      "Open the documentation to see more examples"
```

Reply:
`truncated` means that there was at least 1 example node at the root of the doc comment.

In the following snippet the example is going to be truncated from the tooltip result because it is a direct child of the root.

```xml
<summary>
Create a promise from a function
</summary>
<typeparam name="'T">Return type of the promise</typeparam>
<returns>
The promise created by the function
</returns>
<example>
<code lang="fsharp">
let write (path: string) (content: string) =
    Promise.create (fun resolve reject ->
        Node.Api.fs.writeFile(path, content, (fun res ->
            match res with
            | Some res -> reject (res :?> System.Exception)
            | None -> resolve ()
        ))
    )
</code>
</example>
```

In the following snippet, the example is not truncated because it is part of another node (here the `summary` node).

```xml
<summary>
Create a promise from a function
<example>
<code lang="fsharp">
let write (path: string) (content: string) =
    Promise.create (fun resolve reject ->
        Node.Api.fs.writeFile(path, content, (fun res ->
            match res with
            | Some res -> reject (res :?> System.Exception)
            | None -> resolve ()
        ))
    )
</code>
</example>
</summary>
<typeparam name="'T">Return type of the promise</typeparam>
<returns>
The promise created by the function
</returns>
```

We makes the distinction here, because if we truncate the examples when inside another nodes we are loosing context and depending on how the text was formulated it could makes the tooltip not useful.

Same, if there are examples in different nodes without that, we would just concatenate them at the end of the tooltip but we would not know if they were in the summary, exceptions, returns, etc. nodes.